connector/connect/common/src/main/protobuf/spark/connect/types.proto

```diff
@@ -101,7 +101,7 @@ message DataType {
 
   message String {
     uint32 type_variation_reference = 1;
-    uint32 collation_id = 2;
+    string collation = 2;
```

Comment:
Why are we making this change?

Reply:
This is protobuf collation field used by Spark Connect. We wanted to decouple knowledge about collation ids in Spark Connect client and server in order to make collation ids as internal as possible, thus we're not serializing collation id anymore but collation name instead.

This enables us to forget about collation ids in PySpark and use only collation name which can be an arbitrary string and all logic of parsing/validating will be run on Scala side, so we don't need to have duplicate code.

Reply:
And it makes sense. Collation id should be internal spark concept. Outside of spark you should always use collation name.

Reply:
It took me a bit, but I get it now. Thanks!